code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -35,6 +35,7 @@
 
 /obj/item/melee/baton/Destroy()
 	if(cell?.loc == src)
+		STOP_PROCESSING(SSobj, src)
```

Comment:
I think this should go outside of any conditional, just to be safe. Batons should call stop processing when they're destroyed no matter where the cell is.